Amazon.IonDotnet.Tests/Internals/TextWriterTest.cs

```diff
@@ -81,5 +81,28 @@ public void WriteSymbolWithSymbolTable(String tableSym, String newSym, String ex
             var actual = sw.ToString();
             Assert.AreEqual(expected, actual);
         }
+
+        [TestMethod]
+        [ExpectedException(typeof(UnknownSymbolException))]
+        public void WriteSymbolWithSymbolTableOutOfRange()
+        {
+            var text =
+                SystemSymbols.IonSymbolTable + "::" +
+                "{" +
+                "symbols:[\"s1\"]" +
+                "}\n" +
+                "$10\n";
+
+            var ionValueFactory = new ValueFactory();
+            var datagram = IonLoader.Default.Load(text);
+            datagram.Add(ionValueFactory.NewSymbol("with_unknown_sid"));
+            datagram.Add(ionValueFactory.NewSymbol(new SymbolToken(null, 12)));
+
+            var sw = new StringWriter();
+            var textWriter = IonTextWriterBuilder.Build(sw);
```

Comment:
We need a test (and fix, if necessary) for the binary writer too.

Reply:
Added.